test/e2e/usersignup_test.go

```diff
@@ -515,3 +521,83 @@ func (s *userSignupIntegrationTest) TestSkipSpaceCreation() {
+	_, _ = md5hash.Write([]byte("foo@bar.com"))
+	emailHash := hex.EncodeToString(md5hash.Sum(nil))
+
+	userSignup := &toolchainv1alpha1.UserSignup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "userid-we-migrate-from",
+			Namespace: s.Host().Namespace,
+			Labels: map[string]string{
+				toolchainv1alpha1.UserSignupUserEmailHashLabelKey: emailHash,
+			},
+			Annotations: map[string]string{toolchainv1alpha1.UserSignupUserEmailAnnotationKey: "foo@bar.com"},
+		},
+		Spec: toolchainv1alpha1.UserSignupSpec{
+			Userid:   "userid-we-migrate-from",
+			Username: "username-we-migrate-to",
+		},
+	}
+
+	states.SetApproved(userSignup, true)
+
+	require.NoError(s.T(), s.Awaitilities.Host().Client.Create(context.TODO(), userSignup))
+
+	// Let the UserSignup provision
+	userSignup, err := s.Awaitilities.Host().WaitForUserSignup(userSignup.Name,
+		wait.UntilUserSignupContainsConditions(ApprovedByAdmin()...))
+	require.NoError(s.T(), err)
+
+	// Deactivate
+	states.SetDeactivated(userSignup, true)
+	userSignup, err = s.Awaitilities.Host().UpdateUserSignup(userSignup.Name, func(us *toolchainv1alpha1.UserSignup) {
+		states.SetDeactivated(us, true)
+	})
+	require.NoError(s.T(), err)
+
+	// Wait until the space resources are cleaned up
+	err = s.Awaitilities.Host().WaitUntilSpaceAndSpaceBindingsDeleted(userSignup.Status.CompliantUsername)
+	require.NoError(s.T(), err)
+
+	// The UserSignup should be migrated, so it is expected that a new UserSignup will be created
+	migrated, err := s.Awaitilities.Host().WaitForUserSignup("username-we-migrate-to",
+		wait.UntilUserSignupHasConditions(ConditionSet(
+			ApprovedByAdmin(),
+			DeactivatedWithoutPreDeactivation(),
+			[]toolchainv1alpha1.Condition{
+				{
+					Type:   "UserMigrated",
+					Status: corev1.ConditionFalse,
+					Reason: "MigrationStarted",
+				}, {
+					Type:   toolchainv1alpha1.UserSignupUserDeactivatingNotificationCreated,
+					Status: corev1.ConditionFalse,
+					Reason: toolchainv1alpha1.UserSignupDeactivatingNotificationUserNotInPreDeactivationReason,
+				}, {
+					Type:   toolchainv1alpha1.UserSignupUserDeactivatedNotificationCreated,
+					Status: corev1.ConditionTrue,
+					Reason: toolchainv1alpha1.UserSignupDeactivatedNotificationCRCreatedReason,
+				}, {
+					Type:   toolchainv1alpha1.UserSignupComplete,
+					Status: corev1.ConditionTrue,
+					Reason: toolchainv1alpha1.UserSignupUserDeactivatedReason,
+				}})...,
+		))
+	require.NoError(s.T(), err)
+	require.NoError(s.T(), s.Awaitilities.Host().WaitUntilUserSignupDeleted(userSignup.Name))
+
+	// Now try to reactivate the migrated UserSignup again
+	_, err = s.Awaitilities.Host().UpdateUserSignup(migrated.Name, func(us *toolchainv1alpha1.UserSignup) {
+		states.SetApproved(us, true)
+	})
+	require.NoError(s.T(), err)
+
+	// Confirm that the migrated UserSignup is provisioned ok
+	VerifyResourcesProvisionedForSignup(s.T(), s.Awaitilities, migrated, "base", "base")
+}
```

Comment:
Since we modify the status conditions of the UserSignup, maybe at this point we should test that deactivating and reactivating still works as expected?

Reply:
reactivation of the UserSignup is exactly the purpose of the last few lines, or am I missing something?

Reply:
We deactivate the `userid-we-migrate-from` usersignup and then check that migration happens then reactivate the `username-we-migrate-to` usersignup but we don't check that the `username-we-migrate-to` usersignup can still be deactivated and reactivated. I'm not really sure it's needed so just asking the question of whether it would provide any value

Reply:
ah, I see - so you wanted to do another round of reactivation... I guess that one round is enough - when the user is reactivated it more or less resets all previous conditions... But I can miss some small detail.

Reply:
yes, that's correct. Ok if the conditions would be the same after reactivation then it's not necessary 👍